### PR TITLE
Version 5, SVG, CLI improvements

### PR DESCRIPTION
**Pull request by th-we**
Looking at Rust again after my first attempts at C3D2 Rust meetup 5 years ago.  To practice I hacked away on the "most advanced" branch I could find that also has changes by @davidrusu and @nwoeanhinnogaehr.

This pull request would supersede #5 because it was developed on top of that branch.  Too many things changed to turn everything into separate feature-based pull request.  This therefore became quite large – sorry!

Changes by @davidrusu, @nwoeanhinnogaehr and me are:

* Parser for version 5
* SVG output
* extension of the command line interface including:
  * Colors and cropping (for SVG)
  * Piping out (at least for SVG) and piping in
  * auto-detection of output format or explicit choice of output format
  * no panicking

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,13 +1,14 @@
 [package]
 name = "lines-are-rusty"
 version = "0.1.0"
```
Comment (davidrusu) [inline Cargo.toml]: Let's bump the version to `0.2.0`

**Review by davidrusu — APPROVED** [review]
Good stuff! let's just bump the version and we can release this

**Comment by PeterGrace** [comment]
The SVG rendering on this PR looks correct, but the pdf is different, specifically, the svg retains the calligraphy font but the pdf does not.  Is this expected?

See attached zip file for the svg and pdf output.  Apologies for my disgusting handwriting. :) 
[testpage.zip](https://github.com/ax3l/lines-are-rusty/files/5784909/testpage.zip)

**Comment by th-we** [comment]
@PeterGrace The PDF renderer is really rudimentary at this point.

@davidrusu I bumped the version. Are you co-maintainer of this repo?

**Comment by davidrusu** [comment]
@th-we yep, and thanks, I'll put out a release

**Comment by davidrusu** [comment]
@th-we Do you intend to help maintain this library long-term? If so we can look at adding you as a maintainer as well.

**Comment by th-we** [comment]
@davidrusu Not sure about long-term, but mid-term I definitely would do more.  My motivation (besides practicing Rust) was to import and export annotated PDFs (with proper PDF annotations, not just baked-in lines).  This also requires that entire notebooks can be handled, not just singe .rm files.

I've started working on [XFDF export](https://github.com/th-we/lines-are-rusty/commits/xfdf-organized).  (XFDF can be used to import and export PDF annotations.)

**Comment by davidrusu** [comment]
Very cool, ok if you keep contributing then it might be worth adding you as another co-maintainer. We can deal with that once you've put up the XFDF PR

**Comment by th-we** [comment]
I took quite some liberty to restructure some code. I hope that is O.K. – not sure if there's anybody else working actively on the code right now.

I'm also not sure when I'll find time to do more.  I'm eager to, but there is some danger that I take working on the code as an excuse for procrastinating actually annotating those PDFs ;-).  (Maybe I manage to give both some share of my time.)

**Comment by davidrusu** [comment]
Personally, I welcome refactors, send them in! (but please send them earlier than later, refactors getting out of date are no fun to deal with)

If what you've got provides some incremental value towards annotating PDF's, please just put up a PR and we can get it merged. Someone else will come along can build on top of your work (similar to how you built on top of my rudimentary v5 support and I built on top of @ax3l's initial approach)

**Pull request merged**